util/movement-types/src/lib.rs

```diff
@@ -22,34 +22,44 @@ impl Id {
 	}
 }
 
+impl AsRef<[u8]> for Id {
+	fn as_ref(&self) -> &[u8] {
+		&self.0
+	}
+}
+
 impl fmt::Display for Id {
 	fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
 		write!(f, "{:?}", &self.0)
 	}
 }
 
 #[derive(Serialize, Deserialize, Clone, Default, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
-pub struct Transaction(pub Vec<u8>);
+pub struct Transaction {
+	pub data: Vec<u8>,
+	pub sequence_number: u64,
```

Comment:
The sequence number field could use some documentation: what are the uniqueness and monotonicity requirements?

Reply:
Those would be defined per unit. For just the sequencer, it's basically viewed as merely another means for ordering. `maptos` requires consecutiveness. But, I wasn't planning on this container itself holding those opinions.